platooning_strategic/src/platoon_strategic.cpp

```diff
@@ -974,6 +1028,27 @@ namespace platoon_strategic
             pm_.memberUpdates(vehicleID, platoonId, msg.header.sender_bsm_id, statusParams, dtd);
             ROS_DEBUG_STREAM("Received platoon status message from " << msg.header.sender_id);
         }
+        else if(isJoinRequirementsMsg && msg.header.recipient_id == config_.vehicleID) {
+            has_received_join_requirements_ = true;
+
+            // JOIN_REQUIREMENTS message uses params string format "JOIN_REQUIREMENTS|LANE_INDEX:xx"
+            std::vector<std::string> inputsParams;
+            boost::algorithm::split(inputsParams, strategyParams, boost::is_any_of(","));
+
+            std::vector<std::string> target_lane_index_parsed;
+            boost::algorithm::split(target_lane_index_parsed, inputsParams[0], boost::is_any_of(":"));
+            int target_lane_index = std::stoi(target_lane_index_parsed[1]);
+
+            ROS_DEBUG_STREAM("Received JOIN_REQUIREMENTS MobilityOperation with target lane index: " << target_lane_index);
+
+            // Store the target CandidateFollower platoon lane index provided by the target Leader
+            cf_target_lane_index_ = target_lane_index;
+
+            // Set flag to indicate CandidateFollower must change lanes if it is not currently in the target lane platoon lane index
+            if (current_lane_index_ != cf_target_lane_index_) {
```

Comment:
I don't have a perfect understanding of the lanelet 2 library, but is it possible (if the two negotiating vehicles are on different lanelets for example, where the number of lanes might change) for an index based comparison to have issues?

Reply:
It is possible for an issue to arise in situations where the two negotiating vehicles are on different lanelets, with both lanelets belonging to a lane group of nonequal sizes. I've added a note in the code to document this, and also added a WARN log statement in order to capture when a rear vehicle receives a target lane index from a front vehicle that is located in a different sized lane group than the rear vehicle's current lane group.